test/s3select_test_json_parser.cpp

```diff
@@ -0,0 +1,561 @@
+SUQiOiAiU0dNTCIsCgkJCQkJIlNvcnRBcyI6ICJTR01MIiwKCQkJCQkiR2xvc3NUZXJtIjogIlN0\
+YW5kYXJkIEdlbmVyYWxpemVkIE1hcmt1cCBMYW5ndWFnZSIsCgkJCQkJIkFjcm9ueW0iOiAiU0dN\
+TCIsCgkJCQkJIkFiYnJldiI6ICJJU08gODg3OToxOTg2IiwKCQkJCQkiR2xvc3NEZWYiOiB7CiAg\
+ICAgICAgICAgICAgICAgICAgICAgICJwYXJhIjogIkEgbWV0YS1tYXJrdXAgbGFuZ3VhZ2UsIHVz\
+ZWQgdG8gY3JlYXRlIG1hcmt1cCBsYW5ndWFnZXMgc3VjaCBhcyBEb2NCb29rLiIsCgkJCQkJCSJH\
+bG9zc1NlZUFsc28iOiBbIkdNTCIsICJYTUwiXQogICAgICAgICAgICAgICAgICAgIH0sCgkJCQkJ\
+Ikdsb3NzU2VlIjogIm1hcmt1cCIKICAgICAgICAgICAgICAgIH0KICAgICAgICAgICAgfQogICAg\
+ICAgIH0KICAgIH0KfQoK"
+
+#define TEST5 \
+"ewoKICAgICJnbG9zc2FyeSI6IHsKICAgICAgICAidGl0bGUiOiAiZXhhbXBsZSBnbG9zc2FyeSIsCiAgICAgICAgICAgICAgICAiR2xvc3NEaXYiOiB7CiAgICAgICAgICAgICJ0aXRsZSI6ICJTIi\
+wKICAgICAgICAgICAgICAgICAgICAgICAgIkdsb3NzTGlzdCI6IHsKICAgICAgICAgICAgICAgICJHbG9zc0VudHJ5IjogewogICAgICAgICAgICAgICAgICAgICJJRCI6ICJTR01MIiwKICAgICAgI\
+CAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICJTb3J0QXMiOiAiU0dNTCIsCiAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAiR2xvc3NUZXJtIjogIlN0YW5kYXJk\
+IEdlbmVyYWxpemVkIE1hcmt1cCBMYW5ndWFnZSIsCiAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAiQWNyb255bSI6ICJTR01MIiwKICAgICAgICAgICAgICAgICAgICAgIC\
+AgICAgICAgICAgICAgICAgICJBYmJyZXYiOiAiSVNPIDg4Nzk6MTk4NiIsCiAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAiR2xvc3NEZWYiOiB7CiAgICAgICAgICAgICAg\
+ICAgICAgICAgICJwYXJhIjogIkEgbWV0YS1tYXJrdXAgbGFuZ3VhZ2UsIHVzZWQgdG8gY3JlYXRlIG1hcmt1cCBsYW5ndWFnZXMgc3VjaCBhcyBEb2NCb29rLiIsCiAgICAgICAgICAgICAgICAgIC\
+AgICAgICAgICAgICAgICAgICAgICAgICAgICAgICJHbG9zc1NlZUFsc28iOiBbIkdNTCIsICJYTUwiXSwKICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgInBv\
+c3RhcnJheSI6IHsKICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICJhIjoxMTEsCiAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgIC\
+AgICAgICAgICAgICAgICAgICAgICAgICAiYiI6MjIyCiAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgIH0KICAgICAgICAgICAgICAgICAgICB9LAogICAgICAg\
+ICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgIkdsb3NzU2VlIjogIm1hcmt1cCIKICAgICAgICAgICAgICAgIH0KICAgICAgICAgICAgfQogICAgICAgIH0KICAgIH0KfQo="
+
+std::string run_sax(const char * in)
+{
+	JsonParserHandler handler;
+
+	handler.key_value_criteria = true;
+
+	int status = handler.process_json_buffer(base64_decode(std::string(in)).data(), strlen(in));
+
+	if(status==0)
+	{
+		return handler.get_full_result();	
+	}
+
+	return std::string("failure-sax");
+}
+/*
+std::string run_filter_sax(const char * in, std::vector<std::string>& pattern)
+{
+	JsonParserHandler handler;
+
+
+	//handler.filter_pattern = {"glossary", "GlossDiv", "GlossList", "GlossEntry", "GlossDef", "postarray"};
+
+	handler.filter_pattern = pattern;
+
+	int status = handler.process_json_buffer(base64_decode(std::string(in)).data(), strlen(in));
+
+	if(status==0)
+	{
+		return handler.get_filter_result();	
+	}
+
+	return std::string("failure-sax");
+}*/
+
+std::string run_exact_filter(const char * in, std::vector<std::vector<std::string>>& pattern)
```

Comment:
` std::vector<std::vector<std::string>>` is a correct data-structure for unknown number of required keys(it's path actually)